### ISSUE TITLE
Datagrid: Table pagination accessibility issues

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#2369]
1. The current/active page is not identified as “current page.” Users cannot determine which page is currently selected (screenshot 1)
2. Enter key does not activate the combo box. Space key activates it but displays an empty list.([https://media-cdn.atlassian.com/file/17f6223e-affd-46ea-b4a5-d3587b9298af/artifact/video[…]lZCI6ZmFsc2V9.pReng9mjTL6tAzy37vSc2tOq6s6mQKnjWaMudv_ur6Q](https://media-cdn.atlassian.com/file/17f6223e-affd-46ea-b4a5-d3587b9298af/artifact/video_1280.mp4/binary/cdn?client=f8316f75-1151-484f-860f-73a63cdf679e&collection=&max-age=2592000&token=eyJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJmODMxNmY3NS0xMTUxLTQ4NGYtODYwZi03M2E2M2NkZjY3OWUiLCJhY2Nlc3MiOnsidXJuOmZpbGVzdG9yZTpmaWxlOjE3ZjYyMjNlLWFmZmQtNDZlYS1iNGE1LWQzNTg3YjkyOThhZiI6WyJyZWFkIl19LCJleHAiOjE3NzQzNjEyODcsIm5iZiI6MTc3NDM2MDM4NywiYWFJZCI6IjcxMjAyMDplYWE3Y2UwMy1iYjMxLTQzOTYtYmY1YS00ZDY2YTEzMzY4YjAiLCJodHRwczovL2lkLmF0bGFzc2lhbi5jb20vYXBwQWNjcmVkaXRlZCI6ZmFsc2V9.pReng9mjTL6tAzy37vSc2tOq6s6mQKnjWaMudv_ur6Q)
3. Screen reader should announce selected option along with its selected state. Proper ARIA attributes (aria-selected/aria-activedescendant) should be implemented ([https://media-cdn.atlassian.com/file/17f6223e-affd-46ea-b4a5-d3587b9298af/artifact/video[…]lZCI6ZmFsc2V9.BYBJ9HfDiLnZw5lJNIw6jVZKGuIRFrLLGdQ5b0f2me4](https://media-cdn.atlassian.com/file/17f6223e-affd-46ea-b4a5-d3587b9298af/artifact/video_1280.mp4/binary/cdn?client=f8316f75-1151-484f-860f-73a63cdf679e&collection=&max-age=2592000&token=eyJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJmODMxNmY3NS0xMTUxLTQ4NGYtODYwZi03M2E2M2NkZjY3OWUiLCJhY2Nlc3MiOnsidXJuOmZpbGVzdG9yZTpmaWxlOjE3ZjYyMjNlLWFmZmQtNDZlYS1iNGE1LWQzNTg3YjkyOThhZiI6WyJyZWFkIl19LCJleHAiOjE3NzQzNjEwODgsIm5iZiI6MTc3NDM2MDE4OCwiYWFJZCI6IjcxMjAyMDplYWE3Y2UwMy1iYjMxLTQzOTYtYmY1YS00ZDY2YTEzMzY4YjAiLCJodHRwczovL2lkLmF0bGFzc2lhbi5jb20vYXBwQWNjcmVkaXRlZCI6ZmFsc2V9.BYBJ9HfDiLnZw5lJNIw6jVZKGuIRFrLLGdQ5b0f2me4) )

<img width="1906" height="811" alt="Image" src="https://github.com/user-attachments/assets/4ef7c352-4212-4adf-86f2-8900291e457d" />

https://cengage.slack.com/archives/C8Y4FR63U/p1774360088347239

**NOTE: Contact the team when the issue is resolved.**
